pkg/registry/apis/provisioning/types.go

```diff
@@ -32,6 +32,10 @@ type Repository interface {
 
 	// For repositories that support webhooks
 	Webhook() http.HandlerFunc
+
+	OnCreate(ctx context.Context) error
+	OnUpdate(ctx context.Context, old Repository) error
```

Comment:
```suggestion
	OnUpdate(ctx context.Context, old *provisioning.Repository) error
```

no need to instantiate the old version